pkg/util/log/clog.go

```diff
@@ -930,6 +933,18 @@ func (sb *syncBuffer) rotateFile(now time.Time) error {
 		return err
 	}
 
+	// Redirect stderr to the current INFO log file in order to capture panic
+	// stack traces that are written by the Go runtime to stderr. Note that if
+	// --logtostderr is true we'll never enter this code path and panic stack
+	// traces will go to the original stderr as you would expect.
+	if sb.sev == Severity_INFO &&
+		logging.stderrThreshold > Severity_INFO &&
+		!logging.noStderrRedirect {
+		if err := syscall.Dup2(int(sb.file.Fd()), syscall.Stderr); err != nil {
```

Comment:
I didn't express myself properly. We have places in the code that print directly to stderr (e.g. `start.go`, "graceful shutdown failed"). If the log file is rotated while such a print is ongoing, the print will be split across the old and new file.

Also, please change the code in `cli/sql.go` to duplicate fd 2 to another fd, and use that as `osStderr`. If a user is starting the CLI shell with logging redirected to files, they will still want to see their shell errors on the terminal.